src/testdrive.F90

```diff
@@ -405,6 +539,34 @@ recursive subroutine run_unittest(test, unit, stat)
     !$omp critical(testdrive_testsuite)
     write(unit, '(a)') message
     !$omp end critical(testdrive_testsuite)
+    
+    stdout = ''
+    if (allocated(error)) then
+      stdout = trim(error%message)
+    endif
+    
+    s = index(message, '[')
+    e = index(message, ']')
+    res = message(s+1:e-1)
+    select case (res)
+      case ('FAILED', 'UNEXPECTED PASS')
+        call junitxml_write_testcase_opening_tag(test%name)
+        call junitxml_write_testcase_failure(res)
+        call junitxml_write_testcase_closing_tag(stdout)
+      case ('EXPECTED FAIL', 'PASSED')
+        if (len_trim(stdout) > 0) then  
+          call junitxml_write_testcase_opening_tag(test%name)
+          call junitxml_write_testcase_closing_tag(stdout)
+        else
+          call junitxml_write_testcase_single_tag(test%name)
+        endif
+      case ('SKIPPED')
+        res = res
+      case default
+        write(unit, '(a)') "Error: Unknown test result '" // res // "' in test '" // test%name // "'!"
+        error stop 2
+    end select
```

Comment:
Maybe a suggestion to avoid the issue with I/O in the parallel loop:

Transform the subroutines `junit_write_testcase_*` in functions returning `charachter` variables, concatanate them in one single `character` variable, and write it in a critical section. This will also avoid the incremental issue raised by @awvwgk 

this could be additionally extended to the whole JUnit.xml file, that is the content of the JUnit.xml file will be first saved in a `character` variable, and be written to the file only if completed.

Reply:
Sorry, for not responding earlier. I was on holiday the last weeks.
I looked into the proposal to return a string instead of write it to the file. But I am not sure, whether it is a pragmatic way forward. I see two options:

1. In FORTRAN 2003+ one can use dynamic allocatable strings on every level of the code (see https://stackoverflow.com/questions/14886787/fortran-character-input-at-undefined-length, https://stackoverflow.com/questions/31084087/using-a-deferred-length-character-string-to-read-user-input).
2. In older FORTRAN (< 2003) the string size needs to be managed on gloabl scope to grow with the amount of test output, since the total size of the output is unknown in advance.

Option 1 is efficient and light-weigth, but it would possibly exclude too many applications using FORTRAN < 2003.
Option 2 is laborious with all the book keeping.

Do you favor one of the above options or do see another way?
Please, let me know your thoughts.

Reply:
My suggestion is to use the Option 1 (i.e. `character(len=:), allocatable` variable). Applications written in older Fortran could still use this library, provided that the used compiler support Fortran 2003+.